src/broadcast/broadcast.rs

```diff
@@ -378,7 +380,9 @@ impl<N: NodeIdT> Broadcast<N> {
             return Ok(Step::default());
         }
         // Caller should make sure message is inserted. unwrap should never panic.
-        let p = self.echos.get(self.our_id()).unwrap().clone();
+        // Could also have another error type if it panics.
+        let (_, p) = self.echos.get(self.our_id()).unwrap();
+        let p = p.clone().unwrap();
```

Comment:
We should not assume that we have our own `Echo` here. (See my earlier comment, which probably became outdated.)

Reply:
Will pass `p` as an argument here.